### DISCUSSION TITLE
Does UE pixel streamer work using ue4-docker?

### DISCUSSION THREAD

**Discussion by czsd** [adamrehn/ue4-docker#228] [Q&A]
Does UE pixel streamer work using ue4-docker? I'm a bit confused as some docs specify windows only, but ue4-docker is built with linux.

**Comment by slonopotamus** [comment]
> ue4-docker is built with linux

I'm not sure why you think so. ue4-docker supports both Linux and Windows.

**Reply by czsd** [reply]
The thing that has me confused is the UE docs say:
"Check your OS and hardware - The Pixel Streaming Plugin can only encode video on computers running Windows operating systems, with certain specific types of GPU hardware. For details, see the [Pixel Streaming Reference](https://docs.unrealengine.com/4.27/en-US/SharingAndReleasing/PixelStreaming/PixelStreamingReference)."

I guess how ue4-docker does pixel streaming may differ than what's documented in the UE docs. I'm new to this all and was tasked with getting a pixel streamer container up.

**Reply by TBBle** [reply]
I'm not sure which doc you're quoting from, but the linked [Pixel Streaming Reference](https://docs.unrealengine.com/4.27/en-US/SharingAndReleasing/PixelStreaming/PixelStreamingReference/) says:
> The Pixel Streaming Plugin supports Windows and Linux platforms. OSX is not supported.

And since ue4-docker can produce both Windows containers and Linux containers you have your choice of platform.

You also might find https://unrealcontainers.com/docs/use-cases/pixel-streaming a useful read for how the container images integrate into Pixel Streaming. In fact, you might not even need ue4-docker, if the pre-built images are sufficient for you.

**Reply by czsd** [reply]
I was looking at: https://docs.unrealengine.com/4.27/en-US/SharingAndReleasing/PixelStreaming/PixelStreamingIntro/
It's under "Prerequisites", this is hilarious how it's contradicted in a subpage.

**Reply by TBBle** [reply]
Yeah, looks like this page's note was overlooked when updating for 4.27. Before 4.27, PixelStreaming was indeed Windows-only without pulling in third-party changes.

**Reply by adamrehn** [reply]
@czsd the official container images that ship with Unreal Engine 4.27 include a runtime image specifically for Pixel Streaming. For more details, see the [Containers Overview](https://docs.unrealengine.com/4.27/en-US/SharingAndReleasing/Containers/ContainersOverview/) page of the docs, and for a demo video see the [Cloud Solutions: Unpacked](https://learn.unrealengine.com/course/3773792) course on the Unreal Online Learning site. (For reference, the demo files are under `Engine/Extras/Containers/Examples/PixelStreaming`.)

I've notified the relevant people about the contradiction in the Pixel Streaming docs, so hopefully that should get fixed at some point.
